test/test_tcp.rs

```diff
@@ -578,7 +578,16 @@ fn connect_error() {
     let poll = Poll::new().unwrap();
     let mut events = Events::with_capacity(16);
 
-    let l = TcpStream::connect(&"10.245.111.123:38381".parse().unwrap()).unwrap();
+    // Pick a "random" port that shouldn't be in use.
+    let l = match TcpStream::connect(&"127.0.0.1:38381".parse().unwrap()) {
```

Comment:
There's still no guarantee that this port won't be in use.  How about an RFC5737 address?  Those will never be routable.

Reply:
I'm going to merge this as is, but if you have thoughts on how to improve this test, we can follow up in another PR.